### PR TITLE
TASK-22 - Completed: Creation of the room-creation.page UI

### PR DESCRIPTION
**Pull request by Asafyaco**
Created room-creation page
- Made a form page for create new room 
 - An option to choose the room name 
- Designed create button
- There's an option to choose the room status - If the room status will be private, there's a passoword filed will be display.
- Created a new EndPoint for the category on constants.ts
- Added instance of our AuthenticationService
- An option to choose the category of the game from the DB + Integrate with an API on the EndPoint
- Added button that navigate back to rooms-list page

### CODE REVIEW

File: src/pages/rooms/room-creation/room-creation.page.jsx [hunk 1]
```diff
@@ -1,9 +1,180 @@
...
+  Radio,
+  InputLabel,
+  Select,
+  MenuItem,
+} from "@mui/material";
+import SendIcon from "@mui/icons-material/Send";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { useNavigate } from "react-router-dom";
+import AuthenticationService from "../../../services/authentication.service";
+import * as signalR from "@microsoft/signalr";
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.jsx]: Remove this signalR import. We don't need real-time functionality on this page.

File: src/pages/rooms/room-creation/room-creation.page.jsx [hunk 2]
```diff
@@ -1,9 +1,180 @@
...
 
 const RoomCreationPage = () => {
-    return (
-        <div className="qcg-room-creation-page"></div>
-    );
+  const authenticationService = new AuthenticationService();
+
+  const [roomName, setRoomName] = useState();
+  const [category, setCategory] = useState(" ");
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.jsx]: useState should start with an empty  string with no spaces useState(""). Currently, it's creating a warning for the MUI (check the console).

File: src/pages/rooms/room-creation/room-creation.page.jsx [hunk 3]
```diff
@@ -1,9 +1,180 @@
...
+
+  const [categories, setCategories] = useState([]);
+
+  const [isPublic, setIsPublic] = useState(true);
+  const [password, setPassword] = useState();
+
+  const navigationService = useNavigate();
+
+  const hubConnection = new signalR.HubConnectionBuilder()
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.jsx]: remove this line. We don't need hubConnections (singalR).

File: src/pages/rooms/room-creation/room-creation.page.jsx [hunk 4]
```diff
@@ -1,9 +1,180 @@
...
+      .then((result) => {
+        setCategories(result.data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    hubConnection.start().then(() => {
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.jsx]: This has nothing to do with signalR, getAllCategories should be outside of the hubConnection. hub connections should be completely removed from this page.

File: src/pages/rooms/room-creation/room-creation.page.jsx [hunk 5]
```diff
@@ -1,9 +1,180 @@
...
+                </FormLabel>
+                <RadioGroup
+                  row
+                  aria-labelledby="creation-page-radio-buttons-group-label"
+                  name="row-radio-buttons-group"
+                >
+                  <div className="qcg-flex qcg-flex-center">
+                    <FormControlLabel
+                      value="public"
+                      control={<Radio />}
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.jsx:122]: change from <Radio /> to <Radio checked={isPublic} /> so it always starts as checked for the public status.

File: src/pages/rooms/room-creation/room-creation.page.scss [hunk 6]
```diff
@@ -1,3 +1,43 @@
 .qcg-room-creation-page {
-    
-}
\ No newline at end of file
+  font-family: "Segoe UI", Tahoma, Geneva, Verdana, sans-serif;
+  text-transform: uppercase;
+  // overflow: scroll;
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.scss]: remove commented code.

File: src/pages/rooms/room-creation/room-creation.page.scss [hunk 7]
```diff
@@ -1,3 +1,43 @@
...
+  overflow-y: scroll;
+  line-height: normal;
+  height: 80vh;
+
+  .wrapper {
+    height: 100%;
+    width: 80%;
+
+    .creation-wrapper {
+      // overflow: hidden;
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.scss]: remove commented code.

File: src/pages/rooms/room-creation/room-creation.page.scss [hunk 8]
```diff
@@ -1,3 +1,43 @@
...
+      // width: 80%;
+      background-color: white;
+      border-radius: 2em;
+      box-shadow: 0 0 3em rgb(124, 124, 124);
+      margin: 1.5em 0;
+      padding: 1em;
+
+      // .create-button {
+      //   overflow-y: scroll;
+      // }
```
Comment (FadiFarah) [inline src/pages/rooms/room-creation/room-creation.page.scss]: also here.

**Comment by FadiFarah** [comment]
Also, there is an error on the console because of the MUI. Take a look at why it's happening and try to get it solved.

File: src/pages/rooms/room-creation/room-creation.page.jsx (hunk 4 above)
Comment (Asafyaco) [reply src/pages/rooms/room-creation/room-creation.page.jsx]: Incase that, I should delete the all useEffect?

**Comment by Asafyaco** [comment]
Commited updated with fix by comments

**Review by FadiFarah — APPROVED** [review]
Great!

**Pull request merged**